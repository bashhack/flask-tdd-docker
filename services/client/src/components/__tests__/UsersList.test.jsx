import React from 'react';
import { render, cleanup } from '@testing-library/react';

import UsersList from '../UsersList';

afterEach(cleanup);

const users = [
    {
        'email': 'testuser1@example.com',
        'id': 1,
        'username': 'testuser1'
    },
    {
        'email': 'testuser2@example.com',
        'id': 2,
        'username': 'testuser2'
    }
];

it('renders', () => {
    const { asFragment } = render(<UsersList users={users}/>);
    expect(asFragment()).toMatchSnapshot();
});

it('renders a username', () => {
    const { getByText } = render(<UsersList users={users}/>);
    expect(getByText('testuser1')).toHaveClass('username');
    expect(getByText('testuser2')).toHaveClass('username');
});