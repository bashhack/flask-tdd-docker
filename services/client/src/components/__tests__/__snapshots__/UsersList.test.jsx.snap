// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <p
      class="box title is-4 username"
      data-testid="1"
    >
      testuser1
    </p>
    <p
      class="box title is-4 username"
      data-testid="2"
    >
      testuser2
    </p>
  </div>
</DocumentFragment>
`;
